controllers/configurationpolicy_controller.go

```diff
@@ -1216,19 +1215,34 @@ func sortRelatedObjectsAndUpdate(
 		}
 	}
 
-	if len(oldRelated) == len(related) {
-		for i, entry := range oldRelated {
-			if !gocmp.Equal(entry, related[i]) {
-				update = true
-			}
+	if !gocmp.Equal(related, oldRelated) {
+		r.deleteDetachedObj(*plc, related, oldRelated)
+		plc.Status.RelatedObjects = related
```

Comment:
`if update` doesn't need anymore, whenever **related** changed.. it updates to new one